src/graphql/admin/root/mutation/account-add-usd-wallet.ts

```diff
@@ -0,0 +1,56 @@
+import { GT } from "@graphql/index"
+
+import { Accounts } from "@app"
+import { mapError } from "@graphql/error-map"
+import { WalletCurrency } from "@domain/shared"
+import { WalletType } from "@domain/wallets"
+import WalletDetailPayload from "@graphql/admin/types/payload/wallet-detail"
+
+const AccountsAddUsdWalletInput = GT.Input({
+  name: "AccountsAddUsdWalletInput",
+  fields: () => ({
+    accountIds: {
+      type: GT.NonNullList(GT.ID),
+    },
+  }),
+})
+
+const AccountsAddUsdWalletMutation = GT.Field<
+  {
+    input: { accountIds: [string] }
+  },
+  null,
+  GraphQLContextForUser
+>({
+  type: GT.NonNullList(WalletDetailPayload),
+  args: {
+    input: { type: GT.NonNull(AccountsAddUsdWalletInput) },
+  },
+  resolve: async (_, args) => {
+    const { accountIds } = args.input
+    if (accountIds instanceof Error) {
+      return [{ errors: [{ message: accountIds.message }] }]
+    }
+
+    const addWalletResults = await Promise.all(
+      accountIds.map((accountId) =>
+        Accounts.addWalletIfNonexistent({
+          accountId: accountId as AccountId,
+          type: WalletType.Checking,
+          currency: WalletCurrency.Usd,
+        }),
+      ),
+    )
+    const walletDetails = addWalletResults.map((wallet) => {
+      if (wallet instanceof Error) {
+        return { errors: [{ message: mapError(wallet).message }] }
+      }
+
+      return { errors: [], walletDetails: wallet }
```

Comment:
errors is an array, you should include the errors in only 1 array and not create and array of this structure. @samerbuna could you please validate this?

Reply:
yes, I agree. This will result in a response that's an array with objects that have the `errors` array. The resp should be an object, not an array.

Reply:
I will make this change